json/src/de/mod.rs

```diff
@@ -535,4 +535,41 @@ mod tests {
 
         assert!(super::from_value::<InMemDicomObject>(serialized).is_ok());
     }
+
+    #[test]
+    fn can_resolve_nan_and_inf_float() {
+        let serialized = serde_json::json!({
+            "00020011": {
+                "vr": "FL",
+                "Value": [
+                    5492.8545,
+                    5462.5205,
+                    "NaN",
+                    "-inf",
+                    "inf"
+                ]
+            }
+        });
+
+        let obj: InMemDicomObject = super::from_value(serialized).unwrap();
+        let tag = Tag(0x0002, 0x0011);
+        let element = obj.get(tag).unwrap();
+
+        // verify NAN, INFINITY, and NEG_INFINITY are correctly deserialized to f32::NAN, f32::INFINITY, and f32::NEG_INFINITY
+        let actual_values = element.float32_slice().unwrap();
```

Comment:
I wonder what would happen if the consumer were to request 64-bit floats ([`.to_multi_float64`](https://docs.rs/dicom-core/latest/dicom_core/value/enum.Value.html#method.to_multi_float64)). It should hopefully retain these values, but we can probably append some checks to this test.

Reply:
Sounds good, added a test for this